babbage/query/cuts.py

```diff
@@ -18,14 +19,28 @@ def cut(self, ast):
             raise QueryException('Invalid cut: %r' % ast[0])
         self.results.append((ast[0], ast[1], value))
 
+    def _check_type(self, ref, value):
+        model_type = self.cube.model[ref].datatype
+        if type(value) is str and model_type == 'string':
+            return
+        if type(value) is unicode and model_type == 'string':
+            return
+        elif type(value) is int and model_type == 'integer':
+            return
+        elif type(value) is datetime.datetime and model_type == 'date':
+            return
+        else:
+            raise QueryException('Invalid value %r for cut %s of type %s' % (value, ref, model_type))
```

Comment:
I think the exception message should include both the expected type and the type the user passed in. Knowing how we've treated their value will help the user debug.